README.md

```diff
@@ -13,55 +13,35 @@ interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
 The HSP is a simple bidirectional stream based protocol to transmit messages
 between two connected peers.
 
-# Data Types
-
-## VarInt
-
-The *VarInt* (variable length integer) type encodes unsigned integers
-into sequences of bytes.  Each integer has exactly one encoding.
-
-The integer is split into groups of seven bits. Each group makes up the lower
-seven bits of a byte, least significant group first. The most significant bit
-of each byte is set if and only if more bytes follow.
+# Byte Order
 
-To make the encoding unambigious, the last byte MUST NOT be zero, unless it
-is the only byte in the encoding (i.e. the encoding of the value zero). There
-must be at least one byte in the encoding.
+The byte-order of all key-fields MUST be Big-Endian. Byte-order of
+payload-fields MAY be defined by the application.
 
-The application SHOULD impose a maximum value to avoid integer and buffer
-overflows.
+# Signedness
 
-The *VarInt* has the same encoding as the Google protobuf "Base 128 Varint":
+All key-fields MUST be represented unsigned.
+  * *1-Byte*: Unsigned char
+  * *2-Byte*: Unsigned short
```

Comment:
Can't use terms like "short" because they are not clearly defined. E.g. C specifies that the maximum of "unsigned short" is at least 65535. A value range [0, 262143] for an unsigned short would be perfectly fine in C ;-)

```suggestion
  * *2-Byte*: 16 bit unsigned integer
```